compiler/src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -1252,7 +1277,7 @@ class Inliner(call: tpd.Tree, rhsToInline: tpd.Tree)(using Context) {
     }
 
     override def typedIf(tree: untpd.If, pt: Type)(using Context): Tree =
-      typed(tree.cond, defn.BooleanType) match {
+      typed(tree.cond, defn.BooleanType)(using ctx.addMode(Mode.ForceInline)) match {
```

Comment:
`ForceInline` should only be added if `tree.isInline`.

Reply:
Changed